iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/DataUtil.java

```diff
@@ -0,0 +1,204 @@
+     *
+     * For Parquet and ORC partitions, this will read metrics from the file footer. For Avro partitions,
+     * metrics are set to null.
+     *
+     * @param partition partition key, e.g., "a=1/b=2"
+     * @param uri partition location URI
+     * @param format partition format, avro or parquet
+     * @param spec a partition spec
+     * @param conf a Hadoop conf
+     * @param metricsConfig a metrics conf
+     * @return a List of DataFile
+     */
+  public static List<DataFile> listPartition(Map<String, String> partition, String uri, String format,
+                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig) {
+    return listPartition(partition, uri, format, spec, conf, metricsConfig, null);
+  }
+
+    /**
+     * Returns the data files in a partition by listing the partition location.
+     * <p>
+     * For Parquet and ORC partitions, this will read metrics from the file footer. For Avro partitions,
+     * metrics are set to null.
+     * <p>
+     * Note: certain metrics, like NaN counts, that are only supported by iceberg file writers but not file footers,
+     * will not be populated.
+     * @deprecated use org.apache.iceberg.data.DataUtil#listPartition() once Iceberg 0.12 is released.
+     *
+     * @param partition partition key, e.g., "a=1/b=2"
+     * @param uri partition location URI
+     * @param format partition format, avro or parquet
+     * @param spec a partition spec
+     * @param conf a Hadoop conf
+     * @param metricsConfig a metrics conf
+     * @param mapping a name mapping
+     * @return a List of DataFile
+     */
+  @Deprecated
+  public static List<DataFile> listPartition(Map<String, String> partition, String uri, String format,
+                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig,
+                                             NameMapping mapping) {
+    if (format.contains("avro")) {
+      return listAvroPartition(partition, uri, spec, conf);
+    } else if (format.contains("parquet")) {
+      return listParquetPartition(partition, uri, spec, conf, metricsConfig, mapping);
+    } else if (format.contains("orc")) {
+      return listOrcPartition(partition, uri, spec, conf, metricsConfig, mapping);
+    } else {
+      throw new UnsupportedOperationException("Unknown partition format: " + format);
+    }
+  }
+
+  private static List<DataFile> listAvroPartition(Map<String, String> partitionPath, String partitionUri,
+                                                  PartitionSpec spec, Configuration conf) {
+    try {
+      Path partition = new Path(partitionUri);
+      FileSystem fs = partition.getFileSystem(conf);
+      return Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
+              .filter(FileStatus::isFile)
+              .map(stat -> {
+                Metrics metrics = new Metrics(0L, null, null, null);
```

Comment:
Maybe a comment to indicate that this a workaround to set the record count to 0. Even though it's not correct to do so, there's no other way really